src/neuroconv/tools/nwb_helpers/_dataset_configuration.py

```diff
@@ -0,0 +1,195 @@
+    )
+
+
+def _get_dataset_metadata(
+    neurodata_object: Union[TimeSeries, DynamicTable], field_name: str, backend: Literal["hdf5", "zarr"]
+) -> Union[HDF5DatasetConfiguration, ZarrDatasetConfiguration]:
+    """Fill in the Dataset model with as many values as can be automatically detected or inferred."""
+    DatasetConfigurationClass = BACKEND_TO_DATASET_CONFIGURATION[backend]
+
+    candidate_dataset = getattr(neurodata_object, field_name)
+    # For now, skip over datasets already wrapped in DataIO
+    # Could maybe eventually support modifying chunks in place
+    # But setting buffer shape only possible if iterator was wrapped first
+    if not isinstance(candidate_dataset, DataIO):
+        # DataChunkIterator has best generic dtype inference, though logic is hard to peel out of it
+        # And it can fail in rare cases but not essential to our default configuration
+        try:
+            dtype = DataChunkIterator(candidate_dataset).dtype
+        except Exception as exception:
+            if str(exception) != "Data type could not be determined. Please specify dtype in DataChunkIterator init.":
+                raise exception
+            else:
+                dtype = np.dtype("object")
+
+        full_shape = get_data_shape(data=candidate_dataset)
+
+        if isinstance(candidate_dataset, GenericDataChunkIterator):
+            chunk_shape = candidate_dataset.chunk_shape
+            buffer_shape = candidate_dataset.buffer_shape
+        elif dtype != "unknown":
+            # TODO: eventually replace this with staticmethods on hdmf.data_utils.GenericDataChunkIterator
+            chunk_shape = SliceableDataChunkIterator.estimate_default_chunk_shape(
+                chunk_mb=10.0, maxshape=full_shape, dtype=np.dtype(dtype)
+            )
+            buffer_shape = SliceableDataChunkIterator.estimate_default_buffer_shape(
+                buffer_gb=0.5, chunk_shape=chunk_shape, maxshape=full_shape, dtype=np.dtype(dtype)
+            )
+        else:
+            pass  # TODO: think on this; perhaps zarr's standalone estimator?
+
+        location = _parse_location_in_memory_nwbfile(current_location=field_name, neurodata_object=neurodata_object)
+        dataset_name = location.strip("/")[-1]
+        dataset_info = DatasetInfo(
+            object_id=neurodata_object.object_id,
+            object_name=neurodata_object.name,
+            location=location,
+            dataset_name=dataset_name,
+            full_shape=full_shape,
+            dtype=dtype,
+        )
+        dataset_configuration = DatasetConfigurationClass(
+            dataset_info=dataset_info, chunk_shape=chunk_shape, buffer_shape=buffer_shape
+        )
+        return dataset_configuration
+
+
+def get_default_dataset_configurations(
+    nwbfile: NWBFile,
+    backend: Union[None, Literal["hdf5", "zarr"]] = None,  # None for auto-detect from append mode, otherwise required
+) -> Iterable[DatasetConfiguration]:
```

Comment:
They should both be correct, since a generator is iterable?

Also the sent/received syntax is rather annoying (it would need to be `Generator[DatasetConfiguration, None, None]`)

Reply:
Generator more specific than Iterator and more specific than Iterable. 
Correct was the wrong word in my comment. My sugestion was that Generator is the most specific type that I could think of.
I don't think this is very important though, we are using typing (at least right now) as an extra documentation and I think this conveys this well. Up to you.

Reply:
OK, made it a Generator then

The key difference is that you have to call/force iteration on a generator to see what it returns, unlike lists or tuples, etc.